### ISSUE TITLE
Unable to use Polkadot API with Angular

### ISSUE THREAD

**Issue by afeezaziz** [polkadot-js/api#3064]
After installing and trying the simple example provided in the documents, this error comes up:

`
ERROR in ./node_modules/@polkadot/types/generic/Call.mjs 87:38-46
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/types/generic/AccountId.mjs 12:11-19
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/schnorrkel/sign.mjs 16:21-29
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/types/ethereum/AccountId.mjs 10:11-19
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/types/ethereum/AccountId.mjs 14:11-19
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/types/ethereum/LookupSource.mjs 49:335-343
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/nacl/sign.mjs 26:21-29
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/base58/encode.mjs 13:38-46
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/pbkdf2/encode.mjs 8:18-26
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/pbkdf2/encode.mjs 9:18-26
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/keccak/asU8a.mjs 23:51-59
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/keccak/asU8a.mjs 23:130-138
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/blake2/asU8a.mjs 23:40-48
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/blake2/asU8a.mjs 23:56-64
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/blake2/asU8a.mjs 23:96-104
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/ethereum/encode.mjs 20:21-29
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/base32/encode.mjs 15:14-22
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/base64/encode.mjs 12:21-29
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/mnemonic/bip39.mjs 69:18-26
Can't import the named export 'u8aToU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/util-crypto/crypto.mjs 8:9-18
Can't import the named export 'waitReady' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/metadata/decorate/storage/createFunction.mjs 42:19-30
Can't import the named export 'xxhashAsU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/metadata/decorate/storage/createFunction.mjs 42:45-56
Can't import the named export 'xxhashAsU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/metadata/decorate/storage/getHasher.mjs 6:27-38
Can't import the named export 'xxhashAsU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/metadata/decorate/storage/getHasher.mjs 16:19-30
Can't import the named export 'xxhashAsU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/metadata/decorate/storage/getHasher.mjs 17:19-30
Can't import the named export 'xxhashAsU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/metadata/decorate/storage/getHasher.mjs 18:34-45
Can't import the named export 'xxhashAsU8a' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/api/index.mjs 4:0-44
Can't reexport the named export 'Keyring' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/wasm-crypto-wasm/buffer.mjs 6:0-71
Can't reexport the named export 'buffer' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/wasm-crypto-wasm/buffer.mjs 6:0-71
Can't reexport the named export 'sizeCompressed' from non EcmaScript module (only default export is available)
ERROR in ./node_modules/@polkadot/wasm-crypto-wasm/buffer.mjs 6:0-71
Can't reexport the named export 'sizeUncompressed' from non EcmaScript module (only default export is available)
** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **`

**Comment by jacogr** [comment]
It is a packager issue, not a library issue, some previous items -

https://github.com/polkadot-js/common/issues/814

**Comment by Deekor** [comment]
I've read through the linked comments and am still unsure how to solve it with angular.

**Comment by Deekor** [comment]
> Don't have an answer yet, not quite that familiar with Angular (anymore, it has been a while). However have a reproduction up at https://github.com/jacogr/sample-angular
> 
> Basically the builder seems to use the `module` entry in the `package.json` but then doesn't seem to know how to map the includes from there.
> 
> TL;DR Have a reproduction made, but not sure yet which levers are available to pull in the angular config as of yet

That's fantastic thanks for taking a look. I started writing my own angular library for it and as much fun as that's been it'd be really nice if this library just worked 😃

**Comment by jacogr** [comment]
The "publish esm modules alongside cjs" has have some impacts. Weirdly this is the first Angular report (Vue, React has both has at least 1). Anyway, will find a solution one way or another, unlike the other 2 mentioned Googling in this case was less than fruitful up till now.

**Comment by jacogr** [comment]
With my limited Angular knowledge - see this commit: https://github.com/jacogr/sample-angular/commit/80d0044824a8d437ae4cee9ecb44a910106b438e

It results in an block number update in my same project -

![image](https://user-images.githubusercontent.com/1424473/104956327-3a39d080-59cc-11eb-929d-1187e38e20e7.png)

**Comment by afeezaziz** [comment]
This is great @jacogr !  At least we can load the polkadot API and that shall be sufficient enough for me to work on! Thanks a lot!

**Comment by Deekor** [comment]
@jacogr is there decent documentation of the polkadot jsonrpc somewhere I can reference to work on my angular library? more of a learning experience than anything.

**Comment by jacogr** [comment]
The docs is really the Rust code, same with all type structures used. 

The only docs I'm aware of that are there are the ones I generate from the JS mapping from the Rust code, which is basically this - https://polkadot.js.org/docs/substrate/rpc (It does indicate the JSONRPC name and then the in/out types so can be converted in a normal JSON RPC request)

But apart from that nothing else available I am aware of.

**Comment by Deekor** [comment]
@jacogr  Thanks. Do you happen to be in the polkadot discord or anything that I might be able bounce a few questions off you?

**Comment by afeezaziz** [comment]
If this is the case with webpack, is it all frameworks including Ionic(which is using Angular) and React Native has not been tested with this framework?

This is so because I tried to install the package to be used with React Native but faced with the same problem.

**Comment by Deekor** [comment]
I've tested with Ionic (angular) and indeed it fails - but that is just because it's simply angular with a CSS framework on top

**Comment by afeezaziz** [comment]
Yes, I tried to use same scripts as per git commit by @jacogr but unable to proceed.
